ARGUS/ARGUS_UNet/utils/ARGUS_visualization.py

```diff
@@ -0,0 +1,206 @@
+import numpy as np
+import matplotlib.pyplot as plt
+import os
+import itk
+from itk import TubeTK as tube
+import torch
+
+def view_training_image(self, image_num=0):
+    img_name = self.all_train_images[image_num]
+    print(img_name)
+    img = itk.imread(img_name)
+    lbl = itk.imread(self.all_train_labels[image_num])
+    num_plots = 5
+    num_slices = img.shape[0]
+    step_slices = num_slices / num_plots
+    plt.figure(figsize=[20, 10])
+    for s in range(num_plots):
+        slice_num = int(step_slices * s)
+        plt.subplot(2, num_plots, s + 1)
+        plt.imshow(img[slice_num, :, :])
+        plt.subplot(2, num_plots, num_plots + s + 1)
+        plt.imshow(lbl[slice_num, :, :])
+
+def view_training_vfold_batch(self, batch_num=0):
+    with torch.no_grad():
+        for count, batch_data in enumerate(self.train_loader):
+            if count == batch_num:
+                inputs, labels = (batch_data["image"], batch_data["label"])
+                num_images = inputs.shape[0]
+                plt.figure(figsize=[30, 30])
+                for i in range(num_images):
+                    img = inputs[i]
+                    lbl = labels[i]
+                    num_channels = img.shape[0]
+                    for c in range(num_channels):
+                        plt.subplot(
+                            num_images,
+                            num_channels + 1,
+                            i * (num_channels + 1) + c + 1,
+                        )
+                        plt.imshow(img[c, :, :])
+                    plt.subplot(
+                        num_images,
+                        num_channels + 1,
+                        i * (num_channels + 1) + num_channels + 1,
+                    )
+                    plt.imshow(lbl[0, :, :])
+                break
+
+def view_metric_curves(self, vfold_num, run_id=0):
+    model_filename_base = (
+        "./Results/"
+        + self.filename_base
+        + "-"
+        + str(self.num_slices)
+        + "s-VFold-Run"
+        + str(run_id)
+        + "/"
+    )
+    loss_file = model_filename_base + "loss_" + str(vfold_num) + ".npy"
+    if os.path.exists(loss_file):
+        epoch_loss_values = np.load(loss_file)
+
+        metric_file = model_filename_base + "val_dice_" + str(vfold_num) + ".npy"
+        metric_values = np.load(metric_file)
+
+        plt.figure("Train", (12, 6))
+
+        plt.subplot(1, 2, 1)
+        plt.title("Epoch Average Loss")
+        x = [i + 1 for i in range(len(epoch_loss_values))]
+        y = epoch_loss_values
+        plt.xlabel("epoch")
+        plt.plot(x, y)
+        plt.ylim([0.0, 0.8])
+
+        plt.subplot(1, 2, 2)
+        plt.title("Val Mean Dice")
+        x = [2 * (i + 1) for i in range(len(metric_values))]
+        y = metric_values
+        plt.xlabel("epoch")
+        plt.plot(x, y)
+        plt.ylim([0.0, 0.8])
+
+        plt.show()
+
+def view_testing_results_vfold(self, model_type="best", run_id=[0], device_num=0):
```

Comment:
This function will need to be updated to support multiple classes - instead of just artery classes, but I think it is doable.

Reply:
Once we have  a working experiment for PNB/ONSD, we can start making modifications to make it generic across all tasks.